Makefile

```diff
@@ -1,42 +1,64 @@
 platform = $(shell uname -m)
 
-USB=$(shell pkg-config --cflags libusb-1.0) $(shell pkg-config --libs libusb-1.0)
+USB=$(shell pkg-config --cflags --libs libusb-1.0)
+ifeq (,$(USB))
+  $(warning Did not find USB Libraries, you may need to install libusb-dev.)
+  $(error Missing dependencies)
+endif
+
 DEFS = -D_LIN -D_DEBUG -DGLIBC_20
 CFLAGS = -Wall -Wno-psabi -g -O2 -lpthread
+OPENCV = $(shell pkg-config --exists opencv && pkg-config --cflags --libs opencv || pkg-config --exists opencv4 && pkg-config --cflags --libs opencv4)
+
+ifeq (,$(OPENCV))
+  $(warning Did not find any OpenCV Libraries, you may need to install libopencv-dev.)
+  $(error Missing dependencies)
+endif
 
 ifeq ($(platform), armv6l)
-OPENCV = $(shell pkg-config --cflags opencv) $(shell pkg-config --libs opencv)
-CC = arm-linux-gnueabihf-g++
-AR= arm-linux-gnueabihf-ar
-CFLAGS += -march=armv6
-CFLAGS += -lrt
-ZWOSDK = -Llib/armv6 -I./include
+  CC = arm-linux-gnueabihf-g++
+  AR= arm-linux-gnueabihf-ar
+  CFLAGS += -march=armv6
+  CFLAGS += -lrt
+  ZWOSDK = -Llib/armv6 -I./include
 endif
 
 ifeq ($(platform), armv7l)
-# Some distributions may need to use opencv4 and -DOPENCV_C_HEADERS as is done for x86_64
-OPENCV = $(shell pkg-config --cflags opencv) $(shell pkg-config --libs opencv)
-CC = arm-linux-gnueabihf-g++
-AR= arm-linux-gnueabihf-ar
-CFLAGS += -march=armv7 -mthumb
-ZWOSDK = -Llib/armv7 -I./include
+  # Some distributions may need to use opencv4 and -DOPENCV_C_HEADERS as is done for x86_64
+  CC = arm-linux-gnueabihf-g++
+  AR= arm-linux-gnueabihf-ar
+  CFLAGS += -march=armv7 -mthumb
+  ZWOSDK = -Llib/armv7 -I./include
 endif
 
+#------------------------------------------------------------------------------
```

Comment:
small nit: can you remove these `#----` comments? None of the other platform sections have it, and it's pretty clear from the `ifeq` what platform each section tests.